src/main/java/com/mazawrath/beanbot/commands/MinesweeperCommand.java

```diff
@@ -0,0 +1,130 @@
+package com.mazawrath.beanbot.commands;
+
+import com.mazawrath.beanbot.utilities.Points;
+import com.mazawrath.beanbot.utilities.SentryLog;
+import de.btobastian.sdcf4j.Command;
+import de.btobastian.sdcf4j.CommandExecutor;
+import io.sentry.Sentry;
+import org.apache.commons.lang3.StringUtils;
+import org.javacord.api.DiscordApi;
+import org.javacord.api.entity.channel.ServerTextChannel;
+import org.javacord.api.entity.message.Message;
+import org.javacord.api.entity.server.Server;
+import org.javacord.api.entity.user.User;
+
+import java.util.Arrays;
+import java.util.Random;
+
+public class MinesweeperCommand implements CommandExecutor {
+    private Points points;
+
+    public MinesweeperCommand(Points points) {
+        this.points = points;
+    }
+
+    @Command(
+            aliases = {"minesweeper"},
+            usage = "minesweeper [size] [mines]",
+            description = "Creates a minesweeper field with custom size options.",
+            privateMessages = false
+    )
+
+    public void onCommand(String[] args, DiscordApi api, ServerTextChannel serverTextChannel, User author, Server server, Message message) {
+        SentryLog.addContext(args, author, server);
+
+        int size;
+        int mines;
+
+        if (args.length >= 2) {
+            if (StringUtils.isNumeric(args[0]) && StringUtils.isNumeric(args[1])) {
+                size = Integer.parseInt(args[0]);
+                mines = Integer.parseInt(args[1]);
+            } else {
+                serverTextChannel.sendMessage("Invalid numbers.");
+                return;
+            }
+        } else {
+            size = 10;
+            mines = size * size / 6;
+        }
+
+        if (points.removePoints(author.getIdAsString(), api.getYourself().getIdAsString(), server.getIdAsString(), Points.COMMAND_COST)) {
+            serverTextChannel.sendMessage(new MinesweeperField(ensureInRange(size, 1,14), ensureInRange(mines, 1, size*size)).toDiscordString());
+        }
+
+        Sentry.clearContext();
+    }
+
+    private int ensureInRange(int value, int min, int max) {
+        return (value < min) ? min : ((value > max) ? max : value);
+    }
+
+    private static class MinesweeperField {
+        private final char[][] field;
+
+        public MinesweeperField(int size, int mines) {
+            if (size * size < mines) {
+                throw new IllegalArgumentException("too many mines");
+            }
+            this.field = new char[size][size];
+            for (char[] row : field) {
+                Arrays.fill(row, ' ');
+            }
+            placeMines(mines);
+            calculateFields();
+        }
+
+        private void placeMines(int mines) {
+            Random random = new Random();
+            for (int i = 0; i < mines; ) {
```

Comment:
This can potentially become an infinite loop (technically), because you are finishing the loop only when you got [mines] assigned, but what if random keeps generating previous numbers such that field[x][y] == 'B', then you keep iterating forever.

I would recommend generating N numbers then translate to X,Y or something similar:
https://docs.oracle.com/javase/8/docs/api/java/util/Random.html#ints-long-int-int-